pkg/xds/topology/dataplanes.go

```diff
@@ -3,21 +3,39 @@ package topology
 import (
 	"context"
 
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/pkg/core"
 	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
 	"github.com/kumahq/kuma/pkg/core/resources/manager"
 )
 
-// GetDataplanes returs list of Dataplane in provided Mesh and Ingresses (which are cluster-scoped, not mesh-scoped)
+// GetDataplanes returns list of Dataplane in provided Mesh and Ingresses (which are cluster-scoped, not mesh-scoped)
 func GetDataplanes(ctx context.Context, rm manager.ReadOnlyResourceManager, mesh string) (*core_mesh.DataplaneResourceList, error) {
 	dataplanes := &core_mesh.DataplaneResourceList{}
 	if err := rm.List(ctx, dataplanes); err != nil {
 		return nil, err
 	}
 	rv := &core_mesh.DataplaneResourceList{}
 	for _, d := range dataplanes.Items {
+		if err := ResolveAddress(d); err != nil {
+			return nil, err
+		}
 		if d.GetMeta().GetMesh() == mesh || d.Spec.IsIngress() {
 			_ = rv.AddItem(d)
 		}
 	}
 	return rv, nil
 }
+
+func ResolveAddress(dataplane *core_mesh.DataplaneResource) error {
+	ips, err := core.LookupIP(dataplane.Spec.Networking.Address)
```

Comment:
so what happens when it is IP (which is like 99% of environments)? Do we execute syscall trying to resolve IP? Can we have an if here that if the address is an IP then we skip this step?

Reply:
`net.LookupIP` tries to parse IP address inside - https://golang.org/src/net/lookup.go?s=6141:6181#L242 and return